android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/ConfigurationRegistry.kt

```diff
@@ -491,24 +491,56 @@ constructor(
     gatewayModeHeaderValue: String? = null,
     searchPath: String,
     patientRelatedResourceTypes: MutableList<ResourceType>,
+    currentPage: Int = 1,
   ) {
-    val resultBundle =
-      if (gatewayModeHeaderValue.isNullOrEmpty()) {
-        fhirResourceDataSource.getResource(searchPath)
-      } else
-        fhirResourceDataSource.getResourceWithGatewayModeHeader(
-          gatewayModeHeaderValue,
-          searchPath,
-        )
+    val pageSize = 200
+    var nextPage = currentPage
+    var totalItemsProcessed = 0
+
+    while (true) {
```

Comment:
I don't think we need this here, or rather we should refactor to remove it and use a different approach.